Enable NM specific tests

cathay4t opened the pull request:
Changes to production codes:
 * Support convert NM external managed interface to persistent.
 * Do not include absent interface as changed due to DNS changes.
 * Hide veth information is veth peer is in another namespace.
 * Delete OVS port conenction for OVS system interface also.
 * Ignore `00:00:00:00:00:00` bond `ad_actor_system` option.

cathay4t committed:
test: Enable IEEE 802.1x NM specific tests

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
dns: Do not include absent interface for dns changes

When purging DNS which was stored on interface about to absent,
we should not include this interface into changed network state.

Unit test case included.
Integration test case enabled.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
nm: Preserve existing config for external managed interface

When user want to convert external managed configurations to persistent
through desire state:

```yml
---
interfaces:
  - name: eth1
```

We will lose its route configures. This is caused by NetworkManager not
include route information in external managed NmConnection. To fix this
problem:

 * When `Interface.is_up_exist_config()` and NmConnection has external
   flag, we create connection base on current configure
   `NetworkState().get_kernel_iface_with_route()`. The user space
   interface is always created by NetworkManager, hence no fit into this
   use case.

 * Ignore veth config if veth peer is in another namespace.

Integration test case enabled.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Enable SR-IOV NM specific tests

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Enable iproute NM specific test

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Enable veth NM specific test

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Remove NM specific checkpoint tests

Existing integration test cases already covered the checkpoint tests.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: Remove plugin test

Nmstate rust has not plan on support plugin yet.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
nm: Delete OVS port connection also

When OVS system interface marked as absent, we should delete its
OVS port also.

Integration test case enabled.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera commented on tests/integration/nm/ovs_test.py:
```diff
@@ -229,102 +226,6 @@ def _get_parent_uuid_of_interface(iface_name):
-            assertlib.assert_state_match(changed_bridge.state)
-            new_state = statelib.show_only((BRIDGE0, IFACE0))
-            assert (
-                new_state[Interface.KEY][0][OvsDB.OVS_DB_SUBTREE][
-                    OvsDB.EXTERNAL_IDS
-                ]
-                == OVSDB_EXT_IDS_CONF2_STR
-            )
-            assert (
-                new_state[Interface.KEY][1][OvsDB.OVS_DB_SUBTREE][
-                    OvsDB.EXTERNAL_IDS
-                ]
-                == OVSDB_EXT_IDS_CONF1_STR
-            )
-
-    def test_ovs_bridge_remoev_external_ids(self, disable_ovsdb_plugin):
-        bridge = Bridge(BRIDGE0)
-        bridge.set_ovs_db({OvsDB.EXTERNAL_IDS: OVSDB_EXT_IDS_CONF1})
-        bridge.add_internal_port(
-            IFACE0,
-            ipv4_state={InterfaceIPv4.ENABLED: False},
-            ovs_db={OvsDB.EXTERNAL_IDS: OVSDB_EXT_IDS_CONF2},
-        )
-        with bridge.create():
-            changed_bridge = Bridge(BRIDGE0)
-            changed_bridge.set_ovs_db({OvsDB.EXTERNAL_IDS: {}})
-            changed_bridge.add_internal_port(
-                IFACE0,
-                ipv4_state={InterfaceIPv4.ENABLED: False},
-                ovs_db={OvsDB.EXTERNAL_IDS: {}},
-            )
-            changed_bridge.apply()
-            assertlib.assert_state_match(changed_bridge.state)
-            new_state = statelib.show_only((BRIDGE0, IFACE0))
-            assert not new_state[Interface.KEY][0][OvsDB.OVS_DB_SUBTREE][
-                OvsDB.EXTERNAL_IDS
-            ]
-            assert not new_state[Interface.KEY][1][OvsDB.OVS_DB_SUBTREE][
-                OvsDB.EXTERNAL_IDS
-            ]
```
Why are these tests being removed?

cathay4t replied:
Because there is no overlap beetwen ovsdb plugin and NM  plugin, the ovsdb plugin is only for global configure while NM plugin for interface level configure. Current existing test case already has ovsdb tests.
Those deleted test case is testing whether NM plugin can change interface level ovsdb, so not required anymore.

ffmancera commented on rust/src/lib/nm/bond.rs:
```diff
@@ -46,9 +48,11 @@ fn apply_bond_options(
     }
     if let Some(v) = bond_opts.ad_actor_system.as_ref() {
-        nm_bond_set
-            .options
-            .insert("ad_actor_system".to_string(), v.to_string());
+        if v != AD_ACTOR_SYSTEM_USE_BOND_MAC {
+            nm_bond_set
+                .options
+                .insert("ad_actor_system".to_string(), v.to_string());
+        }
```
This is not true anymore. Please, see https://bugzilla.redhat.com/show_bug.cgi?id=1931881

cathay4t replied:
Thanks. Patch changed

cathay4t committed:
test: add test for bond ad_actor_system=00:00:00:00:00:00

The kernel has fixed the support of ad_actor_system=00:00:00:00:00:00

    https://bugzilla.redhat.com/show_bug.cgi?id=1931881

So this patch just enable the integration test case of it.

Signed-off-by: Gris Ge <fge@redhat.com>
